ARCHITECTURE.md

```diff
@@ -51,34 +51,44 @@ https://github.com/replicatedhq/kurl-api
 
 The browser UI for creating and viewing kURL installers.
 
-#### Go API Proxy
+#### Typescript API
 
-This API proxies most requests directly to the Typescript API.
-Additionally, this API interacts with the object store and is responsible for assembling and streaming air-gap bundles back to the end-user.
+https://github.com/replicatedhq/kURL-api/tree/main/src
 
-#### Typescript API
+The API for creating kURL URLs and rendering installers.
+The API accepts a kURL Installer spec and returns a URL including a deterministic hash of the spec for installing a cluster based on that spec, for example https://k8s.kurl.sh/5e61e80.
+The API interacts stores these hashes in its relational database for retrieval.
+Additionally, the API is responsible for [resolving](https://github.com/replicatedhq/kURL-api/blob/f43d0fb1ed770b938dea97de64f39dacdd495c68/src/installers/index.ts#L955-L977) the add-on version ("latest" and ".x") and [rendering](https://github.com/replicatedhq/kURL-api/blob/f43d0fb1ed770b938dea97de64f39dacdd495c68/src/util/services/templates.ts) the spec when a script is requested, for example https://k8s.kurl.sh/5e61e80/install.sh or shorthand https://k8s.kurl.sh/5e61e80.
+The API [reads](https://github.com/replicatedhq/kURL-api/blob/f43d0fb1ed770b938dea97de64f39dacdd495c68/src/util/package/kurl-version.ts) the current kURL version from the object storage bucket (https://kurl-sh.s3.amazonaws.com/dist/VERSION) and uses that version to lookup add-on version information (e.g. https://kurl-sh.s3.amazonaws.com/dist/v2022.04.19-0/supported-versions-gen.json) for spec resolution.
 
-The API for creating and consuming installers. Interacts with the relational database.
+#### Go API Proxy
+
+https://github.com/replicatedhq/kURL-api/tree/main/cmd/server
+
+This API [proxies](https://github.com/replicatedhq/kURL-api/blob/f43d0fb1ed770b938dea97de64f39dacdd495c68/cmd/server/main.go#L66-L69) most requests directly to the Typescript API.
+Additionally, this API is responsible for [assembling and streaming](https://github.com/replicatedhq/kURL-api/blob/f43d0fb1ed770b938dea97de64f39dacdd495c68/cmd/server/main.go#L212-L396) air-gap bundles back to the end-user.
+Air-gap packages are assembled from individual add-on package archives (e.g. https://kurl-sh.s3.amazonaws.com/dist/v2022.04.19-0/rook-1.0.4.tar.gz) and streamed as a single archive back to the end user when requested (e.g. https://k8s.kurl.sh/bundle/5e61e80.tar.gz).
 
 #### Relational Database
 
 Stores installer hashes.
```

Comment:
How we build the hashes ? Could we describe it as well?
Is hashes made from what?